lib/diagnostic.js

```diff
@@ -6,7 +6,16 @@
 // this callback will be passed by the tests to the reduce method, e.g.
 // `let minimum = array.reduce(min);`
 const min = function (previous, current) {
-
+  // Set a dummy variable to store the result for later
+  const result = 0
+  // For each item in a provided array...
+  for (let i = 0, i < array.length; i++) {
```

Comment:
No need to do a loop inside here, since this function will be passed to `reduce` which does this for us.